src/groupBy.ts

```diff
@@ -13,14 +13,14 @@ import { NonEmptyArray, PredIndexedOptional, PredIndexed } from './_types';
  * @indexed
  * @category Array
  */
-export function groupBy<T>(
-  items: readonly T[],
-  fn: (item: T) => PropertyKey
-): Record<PropertyKey, NonEmptyArray<T>>;
+export function groupBy<Item, Key extends PropertyKey>(
```

Comment:
Not related to this PR, but I think we can replace this definition one day:
```
/** types that may be returned by `keyof` */
export type Key = string | number | symbol;
```
with `PropertyKey`.

Reply:
done: 83e04aac